packages/zone/src/durable.js

```diff
@@ -55,31 +58,49 @@ export const detachedDurableStores = attachDurableStores(() =>
  * Create a zone whose objects persist between Agoric vat upgrades.
  *
  * @param {import('@agoric/vat-data').Baggage} baggage
+ * @param {string} [baseLabel]
  * @returns {import('.').Zone}
  */
-export const makeDurableZone = baggage => {
+export const makeDurableZone = (baggage, baseLabel = 'durableZone') => {
   baggage || Fail`baggage required`;
+
+  const attachedStores = attachDurableStores(() => baggage);
+
+  const { makeOnce, wrapProvider } = makeOnceKit(
+    baseLabel,
+    attachedStores,
+    baggage,
+  );
+
   /** @type {import('.').Zone['exoClass']} */
   const exoClass = (...args) => prepareExoClass(baggage, ...args);
   /** @type {import('.').Zone['exoClassKit']} */
   const exoClassKit = (...args) => prepareExoClassKit(baggage, ...args);
   /** @type {import('.').Zone['exo']} */
   const exo = (...args) => prepareExo(baggage, ...args);
 
-  const attachedStores = attachDurableStores(() => baggage);
+  const mapStore = wrapProvider(attachedStores.mapStore);
 
   /** @type {import('.').Zone['subZone']} */
   const subZone = (label, options = {}) => {
-    const subBaggage = provideDurableMapStore(baggage, label, options);
-    return makeDurableZone(subBaggage);
+    const subBaggage = mapStore(label, options);
+    return makeDurableZone(subBaggage, `${baseLabel}.${label}`);
   };
 
   return Far('durableZone', {
```

Comment:
Should durable zones be durable?

Reply:
That's an interesting question, but I don't have a clear grasp of how the durability would bottom out in actual objects.

Reply:
https://github.com/Agoric/agoric-sdk/pull/7971 suggests that zones should not be self storing for now.

https://github.com/Agoric/agoric-sdk/pull/7972 explored what it would take to make stores and zones self-storable, concluding that it is a big enough separate step that it should be postponed.

Reply:
So, resolving.

Reply:
Resolving